### PR TITLE
Scalarization for posteriors, ScalarizedObjective

### PR DESCRIPTION
**Pull request by Balandat**
In some cases we may be interested in some affine function of the modeled outputs. By linearity, the posterior of that function will be a univariate normal (assuming `q=1`). 

This adds some functionality for achieving this for all analytic acquisition functions.

### CODE REVIEW

File: botorch/posteriors/gpytorch.py [hunk 1]
```diff
@@ -99,3 +100,41 @@ def variance(self) -> Tensor:
         if not self._is_mt:
             variance = variance.unsqueeze(-1)
         return variance
+
+
+def affine_transform(
+    posterior: GPyTorchPosterior, weights: Tensor, offset: float = 0.0
```
Comment (eytan) [inline botorch/posteriors/gpytorch.py]: When would we want to use a non-zero offset?

**Comment by eytan** [comment]
@Balandat, AFAIK these types of objectives are almost always referred to as 'scalarizations' or 'weighted sums' that are 'linear combinations' of objectives.  See e.g., https://arxiv.org/pdf/1402.0590.pdf.   Perhaps we can consider naming this `ScalarizedAcquisitionObjective` or `LinearAcquisitionObjective` instead?

**Comment by Balandat** [comment]
@eytan The reason I call if affine here is because it is, in fact, not linear (b/c of the offset term). `ScalarizedAcquisitionObjective` makes a lot of sense, I can change the name to that.

File: botorch/posteriors/gpytorch.py (hunk 1 above)
Comment (Balandat) [reply botorch/posteriors/gpytorch.py]: We could use this for KG to pass in the (negative) current best, which could make the KG implementation somewhat more elegant.

Comment (Balandat) [reply botorch/posteriors/gpytorch.py]: @danielrjiang @VilockLi

**Pull request closed without merging**